connection.py

```
@@ -108,137 +121,264 @@ def log_to_csv(self, sample_data):
+                print("\nNot enough timestamps collected to estimate rate.")
+        except Exception as e:
+            print(f"Error in LSL rate check: {str(e)}")
+
+    def ble_data_handler(self):
+        TARGET_SAMPLE_RATE = 500.0
+        SAMPLE_INTERVAL = 1.0 / TARGET_SAMPLE_RATE
+        next_sample_time = local_clock()
+        
+        while self.running and self.ble_connection:
+            try:
+                if hasattr(self.ble_connection, 'data_available') and self.ble_connection.data_available:
```

Pylint (3.3.7)

[error] Instance of 'Chords_BLE' has no 'data_available' member

(E1101)